### ISSUE TITLE
Bug: Route placeholder `(:any)` does not pass the whole value to controller

### ISSUE THREAD

**Issue by kenjis** [codeigniter4/CodeIgniter4#6947]
### PHP Version

8.1

### CodeIgniter4 Version

`develop` (ab55fe1994210b873d49246ed24a2141ea8a55fa)

### CodeIgniter4 Installation Method

Git

### Which operating systems have you tested for this bug?

macOS

### Which server did you use?

cli-server (PHP built-in webserver)

### Database

n/a

### What happened?

`(:any)` does not pass the matched value, but passes the first segment only.

### Steps to Reproduce

```php
$routes->get('(:any)', 'Home::index/$1');
```

```php
<?php

namespace App\Controllers;

class Home extends BaseController
{
    public function index($any = '')
    {
        dd($any);
    }
}
```

Navigate to http://localhost:8080/abc/def ,
you will see `$any string (3) "abc"`.

### Expected Output

`$any string (3) "abc/def"`

### Anything else?

```php
<?php

namespace App\Controllers;

class Home extends BaseController
{
    public function index($any = '', $next = '')
    {
        dd($any, $next);
    }
}
```

Navigate to http://localhost:8080/abc/def ,
you will see:
```
$any string (3) "abc"
$next string (3) "def"
```

**Comment by kenjis** [comment]
This is not a bug, because this behavior is documented:
https://codeigniter4.github.io/CodeIgniter4/incoming/routing.html#examples

**Comment by kenjis** [comment]
I send a PR to improve the docs: #6960

**Comment by kenjis** [comment]
Closed by #6960
